docs/reference/api/platform/CriticalSectionLock.md

```diff
@@ -1,3 +1,17 @@
 ## CriticalSectionLock
 
+The CriticalSectionLock class provides a mechanism to access a resource without interruption. With the `lock` API, you can enter critical section with interrupts disabled. The `unlock` API is the exit from critical section, and the state of interrupts is restored upon exit. Nesting of critical section is supported, and interrupts are enabled only when we exit from last nested critical section.
+
+CriticalSectionLock class is based on RAII approach. In other words, lock is acquired in constructor and destroyed automatically when out of scope as part of destructor. We do not recommend you use CriticalSectionLock as global or a member of a class because you will enter critical section on object creation, and all interrupts will be disabled.
```

Comment:
Query: Who or what acquires lock in the constructor? Also, who or what destroys lock automatically when it's out of scope? Also, who or what disables all interrupts when you use CriticalSectionLock as global or a member of a class?

Reply:
Constructor acquires the lock. Destructor destroys lock.